semgrep/semgrep/util.py

```diff
@@ -161,6 +162,19 @@ def sub_check_output(cmd: List[str], **kwargs: Any) -> Any:
     return result
 
 
+def manually_search_file(path: str, search_term: str, suffix: str) -> Optional[str]:
+    """
+    Searches a file for the given search term and, if found,
+    returns the first word that contains that search term
+    """
+    fd = open(path, mode="r")
```

Comment:
Do we want to check file exists before reading? If we are in a subdirectory of the git repo .git/config will not exist

Reply:
We might also want to use the with open(...) as f: syntax https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files so python handles closing the file for us

Reply:
fixed

Reply:
woop didn't see the second comment - one min

Reply:
ok fixed 😄